### PR TITLE
feat(footer): v2 layout updates

### LINKED ISSUE #11090: [Footer] v2 layout update
[Updated v2 design spec](https://www.figma.com/file/C5WDkxjlXM26HXGhkCaFW3/Footer-v2?type=design&node-id=0%3A1&mode=design&t=P8cfP99z9FgvG4et-1)

**Layout adjustments:**

- [ ] New overall layout that moves the `locale selector` into the main link section. This also applies to `language selector`. Effects all breakpoints.
- [ ] New style for `locale selector`
- [ ] New locked theme for the entire `footer` – `gray100` instead of the previous `gray90`
- [ ] New layout for adjunct legal links that mirrors the secondary links (overhanging border). Footer links are added left to right, and will create a new row (just like the secondary links).
- [ ] Spacing adjustment for secondary links section.
- [ ] `Micro` version has added IBM 8-bar logo that matches sizing in `masthead` component.

**Default**
Image

Image

**Short**
Image

Image

**Micro (with added logo)**
Image

Image

### PR DESCRIPTION
**Pull request by ariellalgilmore**
### Related Ticket(s)

Closes #11090 

### Description

New layout for adjunct links and moving locale button below
Screenshot 2023-11-08 at 8 53 49 AM

add IBM logo to micro footer
Screenshot 2023-11-08 at 8 54 25 AM

### Changelog

**New**

- refactoring of styles and slots to allow for different layout for all different size variations
- added new ibm logo

### CODE REVIEW

File: packages/web-components/src/components/footer/legal-nav.ts [hunk 1]
```diff
@@ -67,23 +67,35 @@ class C4DLegalNav extends StableSelectorMixin(LitElement) {
...
+    return this.size !== FOOTER_SIZE.MICRO
+      ? html`
+          <nav class="${prefix}--legal-nav">
+            <div class="${this._getTypeClass()}">
+              <ul>
+                <slot></slot>
+              </ul>
+              <slot name="locale"></slot>
+            </div>
+            <div class="${prefix}--adjunct-links__container">
```
Comment (kennylam) [inline packages/web-components/src/components/footer/legal-nav.ts:79]:
```suggestion
          <nav class="${c4dPrefix}--legal-nav">
            <div class="${this._getTypeClass()}">
              <ul>
                <slot></slot>
              </ul>
              <slot name="locale"></slot>
            </div>
            <div class="${c4dPrefix}--adjunct-links__container">
```

Should be our prefix, right?

File: packages/web-components/src/components/footer/legal-nav.ts [hunk 2]
```diff
@@ -67,23 +67,35 @@ class C4DLegalNav extends StableSelectorMixin(LitElement) {
...
+              <ul>
                 <slot name="adjunct-links"></slot>
+              </ul>
+            </div>
+          </nav>
+        `
+      : html`
+          <nav class="${prefix}--legal-nav">
```
Comment (kennylam) [inline packages/web-components/src/components/footer/legal-nav.ts:87]:
```suggestion
          <nav class="${c4dPrefix}--legal-nav">
```

**Review by kennylam — COMMENTED** [review]
Looks great @ariellalgilmore! Just one small change, I think.

File: packages/web-components/src/components/footer/legal-nav.ts (hunk 1 above)
Comment (ariellalgilmore) [reply packages/web-components/src/components/footer/legal-nav.ts]: yeah good call!

**Review by kennylam — APPROVED** [review]
LGTM!

**Comment by oliviaflory** [comment]
@ariellalgilmore this looks really good, a few small updates:

- [ ] Can we update the two adjunct links copy to the following:
   - Updated terms of use
   - React learning technologies  

- [ ] The `mico` footer looks a bit larger than the `short` footer when you view them in the small breakpoint, but they are the same size in the specs, maybe a spacing carry over from v1.

| Micro | short |
| --- | --- |
| Screenshot 2023-11-09 at 8 19 27 PM | Screenshot 2023-11-09 at 8 11 24 PM |


- [ ] I think the legal links look a bit too spaced out compared to the specs, it looks like Rich has them 8px apart to match the links in the default footer
Screenshot 2023-11-09 at 8 16 40 PM
Screenshot 2023-11-09 at 8 17 01 PM

**Comment by ariellalgilmore** [comment]
@oliviaflory updated!!

**Review by RichKummer — COMMENTED** [review]
@ariellalgilmore this looks so good 💪 
Couple of details that could be fixed up:

## Global
- [x] When `@sm`, the last accordion of the footer links doesn't have a border at the bottom.

Screenshot 2023-11-10 at 3 24 17 PM

## Default
- [x] When `@md`, the nav group loses their headings. This looks like the only breakpoint with this issue.
- [ ] Noticed that there's an `8px` extra bit of spacing that looks like it's coming from a hidden Cookie preferences option. It adds a bit extra to the bottom spacing which is otherwise right. I only see this with the bottom of `legal links` not the `adjunct links`.
Screenshot 2023-11-10 at 3 10 29 PM

Screenshot 2023-11-10 at 3 10 40 PM

**Comment by ariellalgilmore** [comment]
Thanks @RichKummer ! just pushed updates, but weird i'm not seeing the issue on md breakpoint where the headers go away 
Screenshot 2023-11-10 at 1 57 27 PM

**Comment by RichKummer** [comment]
@ariellalgilmore super weird with the headings, but I'm seeing them again based on the last updates 🤔 

Everything checks out, but I did notice that for some reason the `micro` version has the links float upwards a bit – these should be centered vertically in the div:

**Current build:**
Screenshot 2023-11-13 at 11 10 02 AM

**Design:**
Screenshot 2023-11-13 at 11 15 46 AM

**Review by RichKummer — APPROVED** [review]
Looks awesome @ariellalgilmore ! 💯 💯 💯

**Pull request merged**